operator/v1/types.go

```diff
@@ -45,6 +45,27 @@ type ClusterManagerSpec struct {
 	// NodePlacement enables explicit control over the scheduling of the deployed pods.
 	// +optional
 	NodePlacement NodePlacement `json:"nodePlacement,omitempty"`
+
+	// DeployOptions contains the options of deploying a cluster-manager
+	// +optional
+	DeployOption DeployOption `json:"deployOptions,omitempty"`
+}
+
+type DeployOption struct {
+	// DeployMode can be Default or ExternalController, ExternalController indicates the controller is deployed in another cluster but not the hub cluster.
+	// +optional
+	// +kubebuilder:default=Default
+	DeployMode string `json:"deployMode,omitempty"`
+
+	// ExternalController represents the config when using the ExternalController mode.
+	// +optional
+	ExternalController ExternalControllerConfig `json:"externalController,omitempty"`
+}
+
+type ExternalControllerConfig struct {
```

Comment:
how can we get the hub kubeconfig with mode `ExternalController`?

Reply:
it can be hardcoded. But we need to doc here. I do not think this will need user to mutate.

Reply:
From the above discussion https://github.com/open-cluster-management-io/api/pull/103#discussion_r743455095 , we may use a hardcode fixed namespace and secret name to provide hubkubeconfig.

Reply:
Does this mean one registration operator only works for one hub cluster?  If we want to deploy multiple `registration-hub` for multiple hub clusters in one cluster(separate namespaces), we should run multiple registration operators, right?